### PR TITLE
Source Plaid: port to Python CDK

### PR DESCRIPTION
**Pull request by devly-tal**
## What
Reformatted plaid to be written in Python instead of Javascript: #3998
Added transaction functionality to the source.

## How

The new Python connector works with the Plaid API package to draw data, then extracts the relevant data to stream, and dumps it as messages.
Balance stream maintains data structure from previous connector version.

## Recommended reading order
Chef's choice. 😄 

## Pre-merge Checklist

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

### CODE REVIEW

**Comment by alafanechere** [comment]
Hi @firmbase-tal , thanks for this contribution! Could you please share evidence that acceptance tests are passing on your side?
Command to run from top level airbyte directory:
```sh
./gradlew :airbyte-integrations:connectors:source-plaid:integrationTest`
```
Please also run: 
```sh
./gradlew bash
```

**Review by alafanechere — CHANGES_REQUESTED** [review]
I made a first quick review of the core and my main concern is that I feel like the abstractions you defined thanks to `configs/PlaidStream/PlaidRequester` are a bit far-fetched. I'm not 100% sure of the interest of storing some streams configurations in an external json file. 
I think that you can make your connector more readable by having the following classes:

* `PlaidStream(Stream)`
* `BalanceStream(PlaidStream)`
* `IncrementalTransactionStream(PlaidStream)`

And IMO the `PlaidRequesters` method could be brought back to the child classes above.

Let me know what you think about this!

**Comment by alafanechere** [comment]
Hi @firmbase-tal, are you still interested in moving this PR forward according to my review?

**Comment by alafanechere** [comment]
I add the label blocked until the user reply the review.

**Comment by firmbase-tal** [comment]
Yes, apologies for the delay, will handle the changes soon.

**Comment by hbd** [comment]
Thanks for your work here @firmbase-tal. Let me know if I can help get this over the line.

File: airbyte-integrations/connectors/source-plaid/acceptance-test-config.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      status: "succeed"
+    - config_path: "integration_tests/invalid_config.json"
+      status: "failed"
+  discovery:
+    - config_path: "secrets/config.json"
+  basic_read:
+    - config_path: "secrets/config.json"
+      configured_catalog_path: "integration_tests/configured_catalog.json"
+      empty_streams: []
+  # TODO uncomment this block to specify that the tests should assert the connector outputs the records provided in the input file a file
```
[the remark this answers is not in the document]
Comment (firmbase-tal) [reply airbyte-integrations/connectors/source-plaid/acceptance-test-config.yml]: > Hi @firmbase-tal , thanks for this contribution! Could you please share evidence that acceptance tests are passing on your side? Command to run from top level airbyte directory:
> 
> ```shell
> ./gradlew :airbyte-integrations:connectors:source-plaid:integrationTest
> ```
> 
> Please also run:
> 
> ```shell
> ./gradlew bash
> ```

```shell
./gradlew bash
```
This fails with error: `Task 'bash' not found in root project 'airbyte'.`
I'm unsure as to what went wrong there.

Also, where should I upload the files from the acceptance tests?

**Comment by firmbase-tal** [comment]
I think I fixed everything for now, please let me know if further changes are necessary.

**Comment by alafanechere** [comment]
Hi @firmbase-tal, thanks for the changes; I'll go for another review this week then! @hbd feel free to review too if you're familiar with Plaid 😄

File: airbyte-integrations/connectors/source-plaid/acceptance-test-config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (alafanechere) [reply airbyte-integrations/connectors/source-plaid/acceptance-test-config.yml]: Sorry it was a typo for my side I meant `./gradlew format`

**Comment by alafanechere** [comment]
@firmbase-tal could you double-check that `docs/integrations/sources/plaid.md` is up to date? Especially the part about access token creation. I tried to renew the one we have but following the instructions led to an error and it seems not to correspond with the [online Plaid documentation](https://plaid.com/docs/api/tokens/)

**Comment by alafanechere** [comment]
@firmbase-tal I allowed myself to make some changes to simplify the code.
Mainly:
* Schemas are usually declared in external json files.
* The `PlaidRequester` class and some abstraction logic looked redundant to me.

Let me know what you think about these.

File: airbyte-integrations/connectors/source-plaid/source_plaid/source.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import plaid
+from airbyte_cdk.logger import AirbyteLogger
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from plaid.api import plaid_api
+from plaid.model.accounts_balance_get_request import AccountsBalanceGetRequest
+from plaid.model.transactions_get_request import TransactionsGetRequest
+
+from .plaid_requester import PlaidRequester
```
Comment (hbd) [inline airbyte-integrations/connectors/source-plaid/source_plaid/source.py:20]: @jrhizor I think this needs to be removed as well if the `PlaidRequester` is going to be removed

**Review by hbd — COMMENTED** [review]
LGTM. I was able to build the image locally and, although I am not very familiar with the Plaid API, the spec looks good

**Review by alafanechere — APPROVED** [review]
I confirm acceptance tests are passing. I'm publishing this now.

**Comment by hbd** [comment]
Just confirming that I ran latest locally and successfully synced Plaid Sandbox data.

**Comment by alafanechere** [comment]
Thank you @hbd for testing and @firmbase-tal for your original contribution!

**Pull request merged**